File: wire-android-automation/common/src/main/java/com/wearezeta/auto/common/calling2/v1/model/InstanceStatus.java
package com.wearezeta.auto.common.calling2.v1.model;

import java.util.Arrays;

public enum InstanceStatus {
	NON_EXISTENT, STARTING, STARTED, STOPPING, STOPPED, LOGIN_FAILED, DESTROYED, ERROR;

	public static boolean isContainedInSubset(InstanceStatus[] subSet,
			InstanceStatus item) {
		for (InstanceStatus status : subSet) {
			if (item == status) {
				return true;
			}
		}
		return false;
	}

	public static InstanceStatus fromString(String s) {
		return Arrays.stream(InstanceStatus.values())
				.filter(x -> x.name().equalsIgnoreCase(s))
				.findFirst()
				.orElseThrow(
						() -> new IllegalArgumentException(String.format("Instance status '%s' is unknown", s))
				);
	}
}

File: tests/testsSupport/src/main/call/models/InstanceStatus.kt
/*
 * Wire
 * Copyright (C) 2025 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 */

package call.models

enum class InstanceStatus {
    NON_EXISTENT,
    STARTING,
    STARTED,
    STOPPING,
    STOPPED,
    LOGIN_FAILED,
    DESTROYED,
    ERROR;

    companion object {
        /**
         * Checks if the given [item] is contained within the provided [subset].
         */
        fun isContainedInSubset(subset: Array<InstanceStatus>, item: InstanceStatus): Boolean {
            return subset.contains(item)
        }

        /**
         * Converts a string to an [InstanceStatus] enum value.
         * Case-insensitive. Throws [IllegalArgumentException] if the string is unknown.
         */
        fun fromString(value: String): InstanceStatus {
            return entries
                .firstOrNull { it.name.equals(value, ignoreCase = true) }
                ?: throw IllegalArgumentException("Instance status '$value' is unknown")
        }
    }
}
